### PR TITLE
Make parameters ordering of `personal_sign` MetaMask-compatible.

### LINKED ISSUE #894: arguments mixed up in personal_sign method
# Problem

It's impossible to sign in into CK

```
:id"24cc03d2-e96e-4e41-9a09-80b69d79955a"
:method"personal_sign"

:args {2 keys}
:data"0x7a8E9Cd60BCa9409C75E8D300fdb5c3e376CFeaa"
:account"0x43727970746f6b697474696573"
:message_id""
```

data and account mixed up

### PR DESCRIPTION
**Pull request by mandrigin**
MetaMask's `personal_sign` has the different argument order than geth's one.

Here we make our version MetaMask (and DApps)-compatible.

Closes #894

### CODE REVIEW

**Comment by dshulyak** [comment]
@mandrigin did you get why they swapped data and address?

i skimmed through the article, but still confused
it is not even compatible with parity https://github.com/paritytech/parity/blob/master/rpc/src/v1/impls/personal.rs#L143

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/paritytech/parity">paritytech/parity</a></strong></div><div>parity - Fast, light, robust Ethereum implementation.</div></blockquote>

File: services/personal/api.go
Comment (gravityblast) [inline services/personal/api.go:65]: if I start a `geth` console I see it as you defined it before:

```
> web3.personal.ecRecover
function()
```

Comment (mandrigin) [reply services/personal/api.go:65]: yeah, that is our `web3.js` vs how MetaMask shows it in their docs.
In reality, people are using something like `sendAsync( "personal_sign", params: [from, data])`

**Comment by mandrigin** [comment]
@dshulyak I have no clue, I'm as surprised as you are... Unfortunately, when the DApps are using `personal_sign` they seem to be using a MetaMask's version of parameters sent.
Maybe we can try guessing which param is which, but that seems... fragile.
I'm blocking this PR until @flexsurfer tests it against DApps.

File: services/personal/api.go
Comment (adambabik) [inline services/personal/api.go:39]: IMO this should be more defensive and do not panic if `rpcParams[0]` is not a string.

File: geth/node/node.go
Comment (adambabik) [inline geth/node/node.go:89]: If we activate Personal Service always now, I wonder if we can remove handlers and define these `personal_*` methods only in the service.

**Review by adambabik — APPROVED** [review]
LGTM but I asked a question if we could simplify the logic if `activatePersonalService` is always called now.

File: geth/node/node.go
Comment (mandrigin) [reply geth/node/node.go:89]: `personal` namespace isn't exported via `CallRPC`, and it shouldn't.
Custom handlers right now serve the exact case to export 2 methods to `CallRPC` that is accessible to DApps.

Anyway, it looks like CK, for example, behaves very inconsistently with parameter flips, so this PR is blocked until we hear from their devs...

Comment (adambabik) [reply geth/node/node.go:89]: Ok, I remember now. We have `sign` and `recover` exposed through handlers and these handlers call original `Personal` service API.

**Comment by mandrigin** [comment]
So we figured out another way to make CK working, closing this PR.

**Pull request closed without merging**
